src/main/java/uk/gov/hmcts/reform/hmc/helper/GetHearingsResponseMapper.java

```diff
@@ -31,36 +32,36 @@ private void setCaseHearings(List<CaseHearingRequestEntity> entities, GetHearing
         List<CaseHearing> caseHearingList = new ArrayList<>();
         for (CaseHearingRequestEntity entity : entities) {
             CaseHearing caseHearing = getCaseHearing(entity);
-            List<HearingResponseEntity> hearingResponses = getHearingResponseEntities(entity, caseHearing);
-            setHearingDaySchedule(caseHearingList, caseHearing, hearingResponses);
+            Optional<HearingResponseEntity> latestHearingResponseOpt = getLatestHearingResponse(entity, caseHearing);
+            latestHearingResponseOpt.ifPresent(latestHearingResponse ->
+                setHearingDaySchedule(caseHearing, latestHearingResponse));
+            caseHearingList.add(caseHearing);
```

Comment:
This was previously getting all response entities and then (see next comment) overwriting the same data over and over. It was also only populating the `caseHearings` only if there was a response present, even though there are fields could still be present from `case_hearing_request`.